xla/service/while_loop_unroller.cc

```diff
@@ -759,6 +759,10 @@ absl::StatusOr<bool> WhileLoopUnroller::Run(
 
   VLOG(3) << "Number of while instructions in the module to unroll: "
           << unrollable_while_ops.size();
+  // Run pass pipeline specified by user immediately before unroll.
+  if (before_unroll_pipeline_) {
```

Comment:
what's the purpose of this being inside the pass? any specific usecase?

Reply:
We have certain use cases for our hardware that will insert custom-calls that need to be run inside this pass immediately before the unroll.